### ISSUE TITLE
Karabiner Elements breaks "shift" and "command" key on iPad connected with Universal Control.

### ISSUE THREAD

**Issue by yqmmm** [pqrs-org/Karabiner-Elements#3035]
Universal control is release in macOS 12.3 on March 14 2022. It allows user to use their keyboard in iPad that is connected with a MacBook. I noticed that when Karabiner Elements is turned on, "shift" and "command" key do not work on the iPad. They still works on the MacBook. When I quit Karabiner or disabled my physical keyboard in "Devices -> Basic Configuration" of Karabiner Elements, "shift" and "command" keys work again on iPad.

I thoght maybe some of my configuration caused this. But even if I create a empty profile, this issue still exists.

**Comment by yqmmm** [comment]
I found this on Reddit with people having the same issue, not sure if they are using Karabiner Elements though.

https://www.reddit.com/r/MacOSBeta/comments/sy1pv7/universal_control_cmd_and_shift_keys_stop_working/

**Comment by gmelon** [comment]
I have the same problem. On Mac, the left Command key works fine, but not on the wirelessly connected iPad.

In addition, there is a problem that English and Korean are repeatedly entered even if I select the input language in Korean and enter the letters while the karabiner is running.

**Comment by yqmmm** [comment]
> I found this on Reddit with people having the same issue, not sure if they are using Karabiner Elements though.
> 
> https://www.reddit.com/r/MacOSBeta/comments/sy1pv7/universal_control_cmd_and_shift_keys_stop_working/

People in this post said that them same issue happens when logitech mouse. So I guess this happens whenever this is some
software modifying keyboard events? Sounds like this this is going to be a hard problem to fix.

**Comment by tekezo** [comment]
I cannot reproduce this issue, unfortunately.
The modifier keys work well between my MacBook/Mac Mini and iPad Air.

Is this issue related with the known issue of Universal Control?
https://developer.apple.com/documentation/macos-release-notes/macos-12_3-release-notes

**Comment by wAuner** [comment]
I'm having the same issue. If Karabiner is running, not even the Macbook's keyboard (M1) is working as it should.
Holding command seems to work, it shows the shortcut overview. However, no shortcut that involves command (e.g. command + t or command + h) or shift seems to work. Once I close Karabiner, it works immediately.

**Comment by wAuner** [comment]
It looks like no key combinations are working, ⌘ and ⇧ are probably just the most common ones where people might notice it. For example ⌃ + space only worked once for me while Karabiner is running.

**Comment by dofine** [comment]
> > I found this on Reddit with people having the same issue, not sure if they are using Karabiner Elements though.
> > https://www.reddit.com/r/MacOSBeta/comments/sy1pv7/universal_control_cmd_and_shift_keys_stop_working/
> 
> People in this post said that the same issue happens when their logitech mouse is connected. So I guess this happens whenever this is some software modifying keyboard events? Sounds like this is going to be a hard problem to fix.

Disconnet my logitech Master MX ( with logioptions+  installed) ,  while Karabiner still running, cmd / shift works fine on iPad.  Karabiner works fine on Mac.
So I guess it's logitech's problem.

**Comment by gmelon** [comment]
> > > I found this on Reddit with people having the same issue, not sure if they are using Karabiner Elements though.
> > > https://www.reddit.com/r/MacOSBeta/comments/sy1pv7/universal_control_cmd_and_shift_keys_stop_working/
> > 
> > 
> > People in this post said that the same issue happens when their logitech mouse is connected. So I guess this happens whenever this is some software modifying keyboard events? Sounds like this is going to be a hard problem to fix.
> 
> Disconnet my logitech Master MX ( with logioptions+ installed) , while Karabiner still running, cmd / shift works fine on iPad. Karabiner works fine on Mac. So I guess it's logitech's problem.

I also turned off the third-party keyboard and tested it (not Logitech) and even though the karabiner is running, shift+cmd works fine with the built-in keyboard.
Is it a bug in macos rather than a logitech problem? Or is it a karabiner bug that occurs only when the karabiner is connected to another keyboard?

**Comment by dofine** [comment]
> I'm having the same issue. If Karabiner is running, not even the Macbook's keyboard (M1) is working as it should. Holding command seems to work, it shows the shortcut overview. However, no shortcut that involves command (e.g. command + t or command + h) or shift seems to work. Once I close Karabiner, it works immediately.

Hi there! I'm using Intel MacBook Pro and everything works fine if I remove MX Master 3. 
Even the Karabiner rules downloaded from website works, like option+L lock screen. Now option+L can lock my iPad!

Also tested with a USB-cable connected external keyboard, works as well.

**Comment by yqmmm** [comment]
I uninstlled Logitech Options and removed my MX Anywhere from bluetooth. Now the situation is that if Karabiner Elements is turned on, I can use modifier keys when my another external bluetooth keyboard (Not Logitech) is not connected. They did not work if the external bluetooth keyboard is connected.

**Comment by annopkomol** [comment]
After I turned off my MX Master3, all modifiers work on my iPad even when Karabiner is running.

**Comment by Rorotto** [comment]
try on the iPad going to settings > general > hardware keyboard > modifier keys ... and try changing the selections it might fix it  attached photo
![IMG_923B4BCC9570-1](https://user-images.githubusercontent.com/92154072/159224584-86e30129-90bf-4303-986a-d48b4afef4e9.jpeg)

**Comment by antiquatedantelope** [comment]
> try on the iPad going to settings > general > hardware keyboard > modifier keys ... and try changing the selections it might fix it attached photo

Unfortunately this did not work for me. Has anyone found a fix?

**Comment by onezhaoyn** [comment]
Accessibility -> turn on Full keyboard control 

then the modifier keys starts to work

**Comment by antiquatedantelope** [comment]
Is there any way to achieve this same thing, but without having the ugly highlighter show up on the iPad? 

Thanks!

**Comment by tunarider** [comment]
I am experiencing this issue.
All Logitech-related devices and apps have been removed.

I’ve discovered a new behavior.

I’m using a 2.4GHz dongle to connect a keyboard to my Mac Studio, and I’m using a Magic Trackpad 2 via Bluetooth.
The device connected via Universal Control is an iPad Pro.

- When I quit Karabiner, modifier key combinations work correctly on iPad.
- When Karabiner is running, modifier key combinations do not work on iPad.
- If Karabiner is running and the trackpad cursor is on the iPad side, modifier key combinations do not work on iPad.
- If Karabiner is running but the trackpad cursor is on the Mac side, modifier key combinations work on iPad.
  - If I click on an app or the background on the iPad using the trackpad and then move the cursor back to the Mac, modifier key combinations start to work on iPad.

Does this behavior give any clues as to what might be causing the issue?

**Comment by BlueQuantumx** [comment]
> I am experiencing this issue. All Logitech-related devices and apps have been removed.
> 
> I’ve discovered a new behavior.
> 
> I’m using a 2.4GHz dongle to connect a keyboard to my Mac Studio, and I’m using a Magic Trackpad 2 via Bluetooth. The device connected via Universal Control is an iPad Pro.
> 
> * When I quit Karabiner, modifier key combinations work correctly on iPad.
> * When Karabiner is running, modifier key combinations do not work on iPad.
> * If Karabiner is running and the trackpad cursor is on the iPad side, modifier key combinations do not work on iPad.
> * If Karabiner is running but the trackpad cursor is on the Mac side, modifier key combinations work on iPad.
>   
>   * If I click on an app or the background on the iPad using the trackpad and then move the cursor back to the Mac, modifier key combinations start to work on iPad.
> 
> Does this behavior give any clues as to what might be causing the issue?

Yes, I can reproduce the behavior. It's singular and I think it can be very helpful to this issue.

**Comment by luxmentis** [comment]
Some more data: I have this problem. If I disconnect my Logitech bluetooth mouse the problem goes away, and comes back when I reconnect the mouse. 

The difference is that I don't have and have never had Logitech's software on this Mac. I do use Steermouse though, including for button mapping.

**Comment by jwhitley** [comment]
FWIW, I *have* a Logitech MX Master 3 mouse, but this problem reproduces for me even when it's not connected. No Logitech software is installed. (macOS Tahoe 25.6.1, M2 Macbook Pro).

**Comment by tekezo** [comment]
The discussion has drifted away from the original topic, so I'll close this for now.

**Comment by jwhitley** [comment]
@tekezo The discussion is about the exact same bug as the original report. This bug, as reported, still exists and having K-E running prevents use of modifier keys on an iPad connected via Universal Control. The comments are all discussing the individual user's repro contexts, about this same bug, for the most part.

Can you explain your reason for closing this more clearly?

**Comment by jwhitley** [comment]
A note to anyone following this issue: it tentatively appears that this bug has resolved with macOS 27 Golden Gate, Karabiner-Elements 16.3.0.  In limited testing, I'm no longer able to reproduce it. 🎊
